app/models/attachment_reference.rb

```diff
@@ -4,4 +4,28 @@ class AttachmentReference < ActiveRecord::Base
   belongs_to :attachment, inverse_of: :attachment_references
 
   delegate :open, :url, :path, to: :attachment
+
+  # Get the name from the file and then further build or find an attachment based on file's SHA256
```

Comment:
this should not mention sha256, just say that this will create or use an existing attachment object? sha256 is an implementation detail of the attachment class